### PR TITLE
upgrade_test: introduced enable_authorization_after_upgrade

### PR DESCRIPTION
**Pull request by amoskong**
Old Scylla doesn't support Authorization, so we can't use `authorizer` to
enable Authorization before & after upgrade.

This patch introduced a new parameter to enable Authorization after upgrade,
and disable Authorization after rollback to old Scylla.

related with PR https://github.com/scylladb/scylla-cluster-tests/pull/938

This PR reference an old parameter `test_sst3`, which is used to enable sst3 format after upgrade, and disable sst3 format after rollback to old Scylla.

### CODE REVIEW

File: sdcm/sct_config.py [hunk 1]
```diff
@@ -672,6 +672,8 @@ class SCTConfiguration(dict):
              help=""),
         dict(name="test_sst3", env="SCT_TEST_SST3",  type=boolean,
              help=""),
+        dict(name="enable_authorization_after_upgrade", env="SCT_ENABLE_AUTHORIZATION_AFTER_UPGRADE",  type=boolean,
```
Comment (fruch) [inline sdcm/sct_config.py]: why not split it into two ? making it future proof, and enabling other Authorizers ?
```yaml
authorization_in_upgrade: 'ScyllaAuthorizer'
remove_authorization_in_rollback: true
```

Comment (amoskong) [reply sdcm/sct_config.py]: fixed.

**Comment by fruch** [comment]
:+1:

**Comment by bentsi** [comment]
@amoskong do we need to backport this to any other branch?

**Comment by fruch** [comment]
@amoskong  if this isn't merge before #972, please make sure to also change the new yaml under `test-case/` folder

**Pull request merged**